tests/NativeIntersectionsTest.php

```diff
@@ -83,6 +83,20 @@ function test(A&B $in): void {
                 'ignored_issues' => [],
                 'php_version' => '8.1'
             ],
+            'nativeTypeIntersectionNamedClasses' => [
+                'code' => '<?php
+                    class Foo {}
+                    class Bar {}
+                    class Baz
+                    {
+                        public function __construct(private Foo&Bar $foobar) {}
+                    }
+                    new Baz(new Foo());
+                    ',
+                'error_message' => 'InvalidIntersectionType',
+                'ignored_issues' => [],
+                'php_version' => '8.1'
+            ],
```

Comment:
Can we add a test for docblock types as well? What happens [here](https://psalm.dev/r/fa2f58ec4d)? If it doesn't show the issue and you prefer to leave that for a future PR just put `SKIPPED-` in front of the test name and I'll make an issue to track it.

Reply:
It doesn't show the issue - and if we did that right now, I think it would probably surface a lot of issues for people who make use of `@var MockObject&SomeNonFinalClass`.

I know that's use in a few places in my company's source code, for example! I guess a decision would need to be made on the desired behaviour.

Reply:
Oh, wait - `MockObject` is an interface. Ignore that..!